llama-rs/src/llama.rs

```diff
@@ -591,54 +630,59 @@ impl LlamaModel {
     }
 
     pub fn inference_with_prompt<'a>(
-        &self,
+        &mut self,
         vocab: &'a GptVocab,
         params: &InferenceParams,
         prompt: &str,
         rng: &mut impl rand::Rng,
+        stop_after_prompt: bool,
         callback: impl Fn(OutputToken<'a>),
     ) {
-        let embd_inp = self.tokenize(vocab, prompt, true);
-        let mut logits = Vec::new();
+        let beginning_of_sentence = self.n_past == 0;
+        let embd_inp = self.tokenize(vocab, prompt, beginning_of_sentence);
+        let mut logits = vec![0.0; self.hparams.n_vocab as usize];
 
-        // determine the required inference memory per token:
+        // Mem per token is initialized at zero. Will be read the first time
+        // llama_eval is called.
         let mut mem_per_token = 0;
-        let _ = self.llama_eval(
-            params.n_threads,
-            0,
-            &[0, 1, 2, 3],
-            &mut logits,
-            &mut mem_per_token,
-        );
```

Comment:
This is one important change that *might* (?) cause issues on the larger models. The way `llama_eval` works is that when you pass mem_per_token as zero, it computes the memory and sets it back for you so you can pass it after successive calls.

But this bogus call caused issues when we were resuming from a cached prompt (because it overwrites some memory tokens). So I decided to simply remove the call. This apparently works just fine, and thinking about it, I'm not even sure what the intent of the original code was, but let's make extra sure just in case :thinking: